tests/e2e/csi_snapshot_basic.go

```diff
@@ -4014,208 +3903,196 @@ var _ = ginkgo.Describe("Volume Snapshot Basic Test", func() {
-		gomega.Expect(err).NotTo(gomega.HaveOccurred())
-		gomega.Expect(*snapshotContent1.Status.ReadyToUse).To(gomega.BeTrue())
+				framework.Logf("Wait till the volume snapshot is deleted")
+				err = waitForVolumeSnapshotContentToBeDeletedWithPandoraWait(ctx, snapc,
+					*volumeSnapshot1.Status.BoundVolumeSnapshotContentName, pandoraSyncWaitTime)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		}()
 
-		framework.Logf("Get volume snapshot ID from snapshot handle")
-		snapshothandle := *snapshotContent1.Status.SnapshotHandle
-		snapshotId := strings.Split(snapshothandle, "+")[1]
+		ginkgo.By("Fetching the username and password of the current vcenter session from secret")
+		var newPassword, originalConf, username, nimbusGeneratedVcPwd, originalPassword string
+		var vsphereCfg e2eTestConfig
+		var secret *v1.Secret
+		var originalVCPasswordChanged bool
 
-		ginkgo.By("Query CNS and check the volume snapshot entry")
-		err = verifySnapshotIsCreatedInCNS(volHandle, snapshotId)
-		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		if vanillaCluster {
+			secret, err = c.CoreV1().Secrets(csiSystemNamespace).Get(ctx, configSecret, metav1.GetOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
-		ginkgo.By("Fetching the username and password of the current vcenter session from secret")
-		secret, err := c.CoreV1().Secrets(csiSystemNamespace).Get(ctx, configSecret, metav1.GetOptions{})
-		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			originalConf = string(secret.Data[vSphereCSIConf])
+			vsphereCfg, err = readConfigFromSecretString(originalConf)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
-		originalConf := string(secret.Data[vSphereCSIConf])
-		vsphereCfg, err := readConfigFromSecretString(originalConf)
-		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			ginkgo.By(fmt.Sprintln("Changing password on the vCenter host"))
+			nimbusGeneratedVcPwd = GetAndExpectStringEnvVar(nimbusVcPwd)
+			username = vsphereCfg.Global.User
+			originalPassword = vsphereCfg.Global.Password
+			newPassword = e2eTestPassword
+			ginkgo.By(fmt.Sprintf("Original password %s, new password %s", originalPassword, newPassword))
+			err = invokeVCenterChangePassword(ctx, username, nimbusGeneratedVcPwd, newPassword, vcAddress, clientIndex)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			originalVCPasswordChanged = true
+			defer func() {
+				if originalVCPasswordChanged {
+					ginkgo.By("Reverting the password change")
+					err = invokeVCenterChangePassword(ctx, username, nimbusGeneratedVcPwd, originalPassword,
+						vcAddress, clientIndex)
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				}
+			}()
+		} else if supervisorCluster {
+			ginkgo.By("Perform password rotation on the supervisor")
+			csiNamespace := GetAndExpectStringEnvVar(envCSINamespace)
+			passwordRotated, err := performPasswordRotationOnSupervisor(client, ctx, csiNamespace, vcAddress)
```

Comment:
VC password reset is diff than supervisor password rotation.. right?

Reply:
Yes, it is different but as discussed I am writing a separate testcase for supervisor password rotation

Reply:
done